dao/src/test/kotlin/repositories/ortrun/DaoOrtRunRepositoryTest.kt

```diff
@@ -451,6 +462,58 @@ class DaoOrtRunRepositoryTest : StringSpec({
         )
     }
 
+    "findRunsBefore should return all runs before a given time" {
+        val refTime = Instant.parse("2025-01-15T08:34:48Z")
+        val finishedTimes = listOf(
+            refTime - 10.seconds,
+            refTime - 1.seconds,
+            refTime,
+            refTime + 1.seconds
+        )
+
+        // Mock the clock to have defined finishedAt times for the test ORT runs.
+        // For each run, the repository queries the clock twice: for the creation and the completion time.
+        val mockTimes = buildList {
```

Comment:
IMO this is unnecessary: I you look at the test I wrote, you can just:

1. create the ORT runs
2. change their states to finished them
3. list the runs with a time before the `finishedAt` of the last run

No need to mock the clock...

Reply:
To be honest, I found your approach with delays to force a later finished time rather problematic.

Reply:
Ah, could you elaborate ?
Do you think it can lead to flakiness or you don't like the test losing time ?

I find mocking a static function of the JVM more brutal and hackish, if I may.

Reply:
Losing time and that you cannot specify an exact time. I also find the mocking solution as implemented here not really thrilling. Ideally, the repository would have a `Clock` instance as parameter, so that time-related operations could be controlled from the outside. But this would require a major refactoring.